### PR TITLE
Correct the tests in the 'factory' package related to get rid from test repositories on GitHub

### PR DESCRIPTION
**Pull request by artaleks9**
### What does this PR do?
* Correct the selenium tests in the _factory_ package to get rid test repositories on GitHub by use the _kohsuke GitHub_ library
* Add the test project _quickstart_ to the resources

### What issues does this PR fix or reference?
#10890

### CODE REVIEW

**Comment by artaleks9** [comment]
ci-test

**Review by SkorikSergey — APPROVED** [review]

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithKeepDirectoryTest.java [hunk 1]
```diff
@@ -48,15 +45,18 @@
...
+  @Inject private TestGitHubRepository testRepo;
 
   private TestFactory testFactoryWithKeepDir;
 
   @BeforeClass
   public void setUp() throws Exception {
-    testFactoryWithKeepDir =
-        testFactoryInitializer.fromUrl(
-            "https://github.com/" + gitHubUsername + "/gitPullTest/tree/master/my-lib");
+    // preconditions - create the test repository
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithKeepDirectoryTest.java]: Test repo has actually created earlier, and here we are adding project into the test repo.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithRootFolderTest.java [hunk 2]
```diff
@@ -51,14 +46,18 @@
...
-  @Inject private Dashboard dashboard;
+  @Inject private TestGitHubRepository testRepo;
 
   private TestFactory testFactoryWithRootFolder;
 
   @BeforeClass
   public void setUp() throws Exception {
-    testFactoryWithRootFolder =
-        testFactoryInitializer.fromUrl("https://github.com/" + gitHubUsername + "/quickstart");
+    // preconditions - create test the repository
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithRootFolderTest.java]: Test repo has actually created earlier, and here we are adding project into the test repo.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithSpecificBranchTest.java [hunk 3]
```diff
@@ -52,25 +55,35 @@
   @Inject private TestWorkspaceServiceClient workspaceServiceClient;
   @Inject private TestProjectServiceClient testProjectServiceClient;
   @Inject private PullRequestPanel pullRequestPanel;
+  @Inject private TestGitHubRepository testRepo;
 
   private TestFactory testFactoryWithSpecificBranch;
 
   @BeforeClass
   public void setUp() throws Exception {
+    // preconditions - create test the repository
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithSpecificBranchTest.java]: Test repo has actually created earlier, and here we are adding project into the test repo.

**Comment by riuvshin** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/719/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/719//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/11019)
docker image: eclipseche/che-server:11019
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

**Comment by artaleks9** [comment]
ci-test

**Comment by riuvshin** [comment]
ci-test build report:
[Build details](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/733/)
[Test report](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/733//Selenium_20tests_20report/)
[selenium tests report data](http://tests.codenvycorp.com/selenium_test_report_data/11019)
docker image: eclipseche/che-server:11019
https://github.com/orgs/eclipse/teams/eclipse-che-qa please check this report.

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithSpecificBranchTest.java [hunk 4]
```diff
@@ -52,25 +55,35 @@
   @Inject private TestWorkspaceServiceClient workspaceServiceClient;
   @Inject private TestProjectServiceClient testProjectServiceClient;
   @Inject private PullRequestPanel pullRequestPanel;
+  @Inject private TestGitHubRepository testRepo;
```
Comment (dmytro-ndp) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/factory/DirectUrlFactoryWithSpecificBranchTest.java]: You can declare it like the follow to create test repo as auxiliary github user:
```
  @Inject
  @Named(AUXILIARY)
  private TestGitHubRepository testAuxiliaryRepo;
```

**Review by dmytro-ndp — APPROVED** [review]

**Pull request merged**